app/controllers/ems_infra_controller.rb

```diff
@@ -56,29 +56,48 @@ def scaling
         end
       end
 
-      begin
-        # Check if stack is ready to be updated
-        update_ready = @stack.update_ready?
-      rescue => ex
-        log_and_flash_message(_("Unable to initiate scaling, obtaining of status failed: %{message}") %
-                                {:message => ex})
+      update_stack(@stack, scale_parameters_formatted, params[:id], return_message)
+    end
+  end
+
+  def scaledown
+    assert_privileges("ems_infra_scale")
+    redirect_to :action => 'show', :id => params[:id] if params[:cancel]
+
+    # Hiding the toolbars
+    @in_a_form = true
+
+    drop_breadcrumb(:name => _("Scale Infrastructure Provider Down"), :url => "/ems_infra/scaling")
+    @infra = ManageIQ::Providers::Openstack::InfraManager.find(params[:id])
+    # TODO: Currently assumes there is a single stack per infrastructure provider. This should
+    # be improved to support multiple stacks.
+    @stack = @infra.direct_orchestration_stacks.first
```

Comment:
Not too bad; it's the same assumption as the one made for scaling up, and OSP-d has no plans to support multiple underclouds for the forseeable future.